src/snapred/backend/recipe/algorithm/CalibrationNormalizationAlgo.py

```diff
@@ -0,0 +1,149 @@
+        self.declareProperty("FocusWorkspace", defaultValue="", direction=Direction.Output)
+        self.declareProperty("SmoothWorkspace", defaultValue="", direction=Direction.Output)
+        self.setRethrows(True)
+        self.mantidSnapper = MantidSnapper(self, name)
+
+    def chopIngredients(self, ingredients: ingredients):
+        self.run = ingredients.run
+        self.backgroundRun = ingredients.backgroundRun
+        self.reductionIngredients = ingredients.reductionIngredients
+        self.smoothDataIngredients = ingredients.smoothDataIngredients
+        self.calibrationRecord = ingredients.calibrationRecord
+        self.calibrantSample = ingredients.calibrantSample
+        self.calibrationWorkspace = ingredients.calibrationWorkspace
+
+    def PyExec(self):
+        ingredients = ingredients(**json.loads(self.getProperty("Ingredients").value))
+        self.chopIngredients(ingredients)
+
+        focusGroups = self.reductionIngredients.reductionState.stateConfig.focusGroups.json()
+        runNumber: str = self.run.runNumber
+        ipts: str = self.run.IPTS
+        rawDataPath: str = ipts + "shared/lite/SNAP_{}.lite.nxs.h5".format(runNumber)
+        backgroundRunNum: str = self.backgroundRun.runNumber
+        backgroundIpts: str = self.backgroundRun.IPTS
+        backgroundRawDataPath: str = backgroundIpts + "shared/lite/SNAP_{}.lite.nxs.h5".format(backgroundRunNum)
+
+        inputWS = self.getProperty("InputWorkspace").value
+        inputWSName = str(inputWS)
+        if inputWS is None:
+            self.mantidSnapper.Load(
+                "Loading file for Normalization InputWS...",
+                Filename=rawDataPath,
+                OutputWorkspace=inputWS,
+            )
+        backgroundWS = self.getProperty("BackgroundWorkspace").value
+        if backgroundWS is None:
+            self.mantidSnapper.Load(
+                "Loading file for Normalization BackgroundWS...",
+                Filename=backgroundRawDataPath,
+                OutputWorkspace=backgroundWS,
+            )
+        # run the algo
+        self.log().notice("Execution of CalibrationNormalizationAlgo START!")
+
+        raw_data = self.mantidSnapper.RawVanadiumCorrectionAlgorithm(
+            "Correcting Vanadium Data...",
+            InputWorkspace=inputWS,
+            BackgroundWorkspace=backgroundWS,
+            CalibrationWorkspace=self.calibrationWorkspace.json(),
+            Ingredients=self.reductionIngredients.json(),
+            CalibrantSample=self.calibrantSample.json(),
+            OutputWorkspace=f"{inputWSName}_raw_data",
+        )
+
+        groupingworkspace = self.mantidSnapper.CustomGroupWorkspace(
+            "Creating Group Workspace...",
+            StateConfig=self.reductionIngredients.reductionState.stateConfig.json(),
+            InputWorkspace=raw_data,
+            OutputWorkspace="CommonRed",
+        )
+
+        for workspaceIndex in range(len(focusGroups)):
```

Comment:
I don't understand what's going on here. RebinRagged should be done after DiffractionFocusing, not before. Also, I'm not sure what is being looped over? is `len(focusGroups)` the length of the list of different groups (e.g. a list like [all,bank,column], length = 3) or is it the number of subgroups in a single group (e.g. 6 in column)?